test/functional/p2p_filter.py

```diff
@@ -69,18 +69,15 @@ def run_test(self):
         filter_address = self.nodes[0].decodescript(filter_node.watch_script_pubkey)['addresses'][0]
 
         self.log.info('Check that we receive merkleblock and tx if the filter matches a tx in a block')
-        filter_node.merkleblock_received = False
         block_hash = self.nodes[0].generatetoaddress(1, filter_address)[0]
         txid = self.nodes[0].getblock(block_hash)['tx'][0]
+        filter_node.wait_for_merkleblock(int(block_hash, 16))
         filter_node.wait_for_tx(txid)
-        assert filter_node.merkleblock_received
 
         self.log.info('Check that we only receive a merkleblock if the filter does not match a tx in a block')
-        with mininode_lock:
-            filter_node.last_message.pop("merkleblock", None)
         filter_node.tx_received = False
-        self.nodes[0].generatetoaddress(1, self.nodes[0].getnewaddress())
-        filter_node.wait_for_merkleblock()
+        block_hash = self.nodes[0].generatetoaddress(1, self.nodes[0].getnewaddress())[0]
+        filter_node.wait_for_merkleblock(int(block_hash, 16))
```

Comment:
style-nit: I slightly prefer the interface where the hash is passed in as a hex string, as opposed to a plain integer. The integer is less type safe and could be confused with other args such as timeout. Also, putting the burden on the caller to always call `int()` makes tests slightly harder to write.

Reply:
I fully agree, but then for consistency reasons all three methods `wait_for_{header,block,merkleblock}` should have the same interface and be changed to take a hex string accordingly? Maybe a good candidate for a follow-up PR (good first issue).

Reply:
At least wait_for_header is only used in one place:

```
$ git grep wait_for_header
test/functional/p2p_invalid_locator.py:                node.p2p.wait_for_header(int(node.getbestblockhash(), 16))
```

so this could be sneaked into this pull (as a separate commit) or left for later. Whatever you prefer.